### PR TITLE
Show "Unread" indicator for local notifications

### LINKED ISSUE #292: Show unread indicator on the bell icon when there are unread notifications
Currently, if there are only notifications marked as unread via the extension (no "native" github notifications), the notifications (bell) icon at the top right does not show a blue circle, but would be nice if it was.

screen shot 2016-09-04 at 7 24 58 pm

See https://github.com/sindresorhus/refined-github/pull/284 for "mark as unread" feature.

### PR DESCRIPTION
**Pull request by AshikNesin**
### Issue
Unread indicator on the bell icon is displayed only for github notification. Not the local notifications that we have bookmarked.

### Fix
Include local notifications too

### Screenshot of Fix
screen shot 2017-05-14 at 7 16 07 pm

### Note
Not sure I've covered all the use cases. If I've missed something, please let me know 😄 

Fixes #292

### CODE REVIEW

File: extension/content.css [hunk 1]
```diff
@@ -605,3 +605,9 @@ div.inline-comment-form .form-actions,
 #refined-github-project-new-link {
 	margin-top: 5px;
 }
+
+/* Display notification bell icon for bookmarked Issues/PR */
+.notification-indicator .mail-status.local-unread{
+	display: inline-block;
```
Comment (jgierer12) [inline extension/content.css]: Better to do this in JS IMO (via [`$(...).show()`](https://api.jquery.com/show/))

File: extension/mark-unread.js [hunk 2]
```diff
@@ -228,12 +228,33 @@ window.markUnread = (() => {
...
+		const $notificationStatus = $($notificationIndicator).find('.mail-status');
+
+		const localNotifications = localStorage.unreadNotifications;
+		let hasLocalNotification = false;
+
+		if (localNotifications !== '[]' && JSON.parse(localNotifications).length > 0) {
+			hasLocalNotification = true;
+		}
+
+		if (hasLocalNotification) {
```
Comment (jgierer12) [inline extension/mark-unread.js]:
```js
if (JSON.parse(localNotifications).length > 0)
```

File: extension/mark-unread.js [hunk 3]
```diff
@@ -228,12 +228,33 @@ window.markUnread = (() => {
...
+			$notificationIndicator.attr('aria-label', 'You have unread notifications');
+			$notificationStatus.addClass('local-unread');
+		} else {
+			$notificationStatus.removeClass('local-unread');
```
Comment (AshikNesin) [inline extension/mark-unread.js]: @jgierer12 Is it okay to do something like this

```js
		let notificationsLabel = '';
		if (hasLocalNotification) {
			notificationsLabel = 'You have unread notifications';
			$notificationStatus.addClass('local-unread');
		} else {
			$notificationStatus.removeClass('local-unread');
			notificationsLabel = ($notificationStatus.hasClass('unread')) ? 'You have unread notifications' : 'You have no unread notifications';
		}
		$notificationIndicator.attr('aria-label', notificationsLabel);
```

Comment (jgierer12) [inline extension/mark-unread.js]: Yeah, I can't think of another way to do it

File: extension/mark-unread.js [hunk 4]
```diff
@@ -228,12 +229,35 @@ window.markUnread = (() => {
 		return $('#user-links a.name img').attr('alt').slice(1);
 	}
 
+	function unreadIndicatorIcon() {
+		const $notificationIndicator = $('.header-nav-link.notification-indicator');
+		const $notificationStatus = $($notificationIndicator).find('.mail-status');
```
Comment (jgierer12) [inline extension/mark-unread.js]: `$notificationIndicator` already is a jQuery object, so you can just do
```js
const $notificationStatus = $notificationIndicator.find('.mail-status');
```

File: extension/mark-unread.js [hunk 5]
```diff
@@ -228,12 +229,35 @@ window.markUnread = (() => {
...
+
+		const localNotifications = localStorage.unreadNotifications;
+		let hasLocalNotification = false;
+
+		if (JSON.parse(localNotifications).length > 0) {
+			hasLocalNotification = true;
+		}
+		let notificationsLabel = '';
+		if (hasLocalNotification) {
```
Comment (jgierer12) [inline extension/mark-unread.js]: No need for a `hasLocalNotification` variable. Just use `JSON.parse(localNotifications).length > 0` directly as the condition of this `if` statement

File: extension/mark-unread.js [hunk 6]
```diff
@@ -247,6 +265,7 @@ window.markUnread = (() => {
 				$(el).parents('.js-notification').removeClass('unread').addClass('read');
 				markRead(el.href);
 			});
+		unreadIndicatorIcon();
```
Comment (jgierer12) [inline extension/mark-unread.js]: Also run in `markNotificationRead`

Comment (AshikNesin) [inline extension/mark-unread.js]: Your are taking about `markNotificationRead` function in `mark-unread.js` ?

If so, unreadIndicatorIcon is already called in it :)

**Comment by AshikNesin** [comment]
@jgierer12 The code that you're refactored looks much cleaner. Will try to follow it here after 😄

File: extension/mark-unread.js (hunk 6 above)
Comment (jgierer12) [inline extension/mark-unread.js]: Sorry, must have overlooked that 😅

Comment (AshikNesin) [inline extension/mark-unread.js]: @jgierer12 No problem 😄

**Comment by paulmolluzzo** [comment]
Sry, waiting on some notifications to test. I just cleared all of mine. 😄

**Comment by AshikNesin** [comment]
@paulmolluzzo Just for testing notifications, I've created a dummy github account 😜

**Pull request merged**